src/twisted/python/failure.py

```diff
@@ -366,20 +369,17 @@ def __init__(self, exc_value=None, exc_type=None, exc_tb=None, captureVars=False
         #   catching means tracebacks generated here don't tend to show
         #   what called upon the PB object.
 
-        while f:
-            if captureVars:
-                localz = f.f_locals.copy()
-                if f.f_locals is f.f_globals:
-                    globalz = {}
-                else:
-                    globalz = f.f_globals.copy()
-                for d in globalz, localz:
-                    if "__builtins__" in d:
-                        del d["__builtins__"]
-                localz = localz.items()
-                globalz = globalz.items()
+        while captureVars and f:
```

Comment:
can we have this linked to `debug` ?

I did a quick search for `caputeVars` in `src/twisted` and I can see that `caputeVars` is only used in `src/twisted/internet/defer.py` and always linked to `Deferred.debug`

Reply:
Because captureVars is set by `Deferred.debug`, that's implicitly what's going on here already.

Reply:
Thanks. It looks like the `captureVars` argument is kind of a private argument, designed to support `Deferred.debug`.

All good.